tsconfig.json

```diff
@@ -0,0 +1,52 @@
+{
```

Comment:
I like the opinions here. 

For ease of maintenance in the future, we could instead extend one of the `@tsconfig` base configs - https://github.com/tsconfig/bases - what do you think?

Reply:
Probably, I just pulled this from my own lib and made a couple of modifications lol

Reply:
In general I agree. The advantage of those particular ones are:

* Now you don't need to do a close read every time you pick up a new repo
* I feel more confident that it's definitely compatible with the node version we want

Yours looks fine to me though.

Reply:
I'll keep this comment open, but during development, I'll stick to a custom-curated `tsconfig.json` for ease of debugging. Once we know that the refactor and its accompanying `tsconfig.json` are all working as intended, we can look for a combination of base configs to extend to provide the same resulting config.